src/examples/specgan_example/specgan_example.py

```diff
@@ -0,0 +1,274 @@
+
+    # This runs at the start of every epoch
+    def on_epoch_start(self, epoch, step):
+        self.gen_loss_avg = tf.keras.metrics.Mean()
+        self.disc_loss_avg = tf.keras.metrics.Mean()
+
+    # This runs at every step in the training (for each batch in dataset)
+    def on_step(self, step, stats):
+        gen_loss, disc_loss = stats
+        self.gen_loss_avg(gen_loss)
+        self.disc_loss_avg(disc_loss)
+        if step % 100 == 0:
+            print(
+                f"Step: {step}, Gen Loss: {self.gen_loss_avg.result()}, Disc Loss: {self.disc_loss_avg.result()}"
+            )
+
+    # This runs at the end of every epoch and is used to display metrics
+    def on_epoch_complete(self, epoch, step, duration):
+        # display.clear_output(wait=True)
+        print(
+            f"Epoch: {epoch}, Step: {step}, Gen Loss: {self.gen_loss_avg.result()}, Disc Loss: {self.disc_loss_avg.result()}, Duration: {duration} s"
+        )
+        self.generate_and_save_images_epoch(epoch, step)
+
+    def generate_and_save_images_epoch(self, epoch, step):
+        generated = self.generator(self.seed, training=False)
+
+        fig = plt.figure(figsize=(4, 4))
+
+        for i in range(generated.shape[0]):
+            plt.subplot(4, 4, i + 1)
+            plt.imshow(generated[i, :, :, 0])
+            plt.axis("off")
+
+        if "save_images" in self.hparams and self.hparams["save_images"]:
+            if self.image_save_dir is None:
+                raise Exception(
+                    "Could not save image, no save_dir was specified in hparams."
+                )
+            plt.savefig(
+                os.path.join(
+                    self.image_save_dir,
+                    "image_at_epoch_{:04d}_step_{}.png".format(epoch, step),
+                )
+            )
+
+        if self.hparams["plot"]:
+            plt.show()
+
+    def sample_sound(self, seed, pipeline):
+        generated = self.generator(seed, training=False)
+
+        print("Generated melspec samples:")
+        fig = plt.figure(figsize=(4, 4))
+
+        for i in range(generated.shape[0]):
+            plt.subplot(4, 4, i + 1)
+            plt.imshow(generated[i, :, :, 0])
+            plt.axis("off")
+
+        if self.hparams["plot"]:
+            plt.show()
+
+        return pipeline(tf.unstack(generated))
+
+
+if __name__ == "__main__":
+    # Load hyperparams from yaml file
+    defaults = load_hparams("hparams.yaml")
+
+    parser = argparse.ArgumentParser(description="Start training of the gan_example")
+    parser.add_argument(
+        "--plot", help="Enable to activate plotting", action="store_true",
+    )
+    parser.add_argument(
+        "--saveimg", help="Enable to save images after each epoch", action="store_true",
+    )
+    parser.add_argument(
+        "--save_dir",
+        metavar="PATH",
+        help="Set the save directory for images and checkpoints",
+        type=str,
+        nargs="?",
+        const=defaults["save_dir"],
```

Comment:
`const` is only used with `store_const` and `append_const`. You want `default`. (Same on many of the others below.)
```suggestion
        default=defaults["save_dir"],
```